web/src/beta/lib/core/Map/Layers/hooks.ts

```diff
@@ -365,7 +365,8 @@ export default function useHooks({
         rawLayer.type === "simple" &&
         rawLayer.data?.value &&
         // If data isn't cachable, reuse layer id for performance.
-        DATA_CACHE_KEYS.some(k => !rawLayer.data?.[k])
+        DATA_CACHE_KEYS.some(k => !rawLayer.data?.[k]) &&
+        Object.isExtensible(rawLayer.data.value)
```

Comment:
User need to take care not to pass inextensible object if some problem occurs later.